### PR TITLE
Update changelog for 21.0

### PR DESCRIPTION
**Pull request by pradyunsg**
I forgot to do this, when I made the release. :)

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -9,8 +9,18 @@ No unreleased changes.
 21.0 - 2021-07-03
 ~~~~~~~~~~~~~~~~~
 
-* `packaging` is now only compatible with Python 3.6 and above.
```
Comment (uranusjr) [inline CHANGELOG.rst:12]: I don't think this line should be removed?

Comment (pradyunsg) [reply CHANGELOG.rst:12]: It's reintroduced with more granularity below:

```
* Drop support for Python 2.7, Python 3.4 and Python 3.5.
```

**Comment by pradyunsg** [comment]
Looks like our required workflows are skippable:

https://github.com/pypa/packaging/blob/d59efa2141210447ae21c4d48813d720e0d1dbfd/.github/workflows/lint.yml#L4-L9

:/

**Comment by pradyunsg** [comment]
Merging since I don't imagine this would break anything and I've re-read this thrice now. :)

**Comment by henryiii** [comment]
Could you make a "GitHub Release" so that the version GitHub shows is correct (and so release watchers will be notified)? I forgot that 21 was out and saw the "20.9" in the sidebar, so I thought this was going to be released, not already released.

**Comment by pradyunsg** [comment]
Whelp. I really goofed up with this release.

Done now: https://github.com/pypa/packaging/releases/tag/21.0

**Comment by henryiii** [comment]
Thanks!

**Pull request merged**